fix(style): button refactoring

platon-rov opened the pull request:
## Related Issue

Closes none.

## Description

Button components refactoring.

## Screenshots

Chromatic detects many changes but they're actually not relevant.

N1XUS commented on src/styles/button.scss:
```diff
@@ -17,34 +11,26 @@ $fd-button-height: 2.25rem;
-$fd-menu-button-max-width: 12rem;
-
-$block: #{$fd-namespace}-button;
+.#{$block-button} {
```
I think we use just `$block` naming for the root selector

platon-rov replied:
file where this variable is defined is imported by the many other files, it's done to avoid any collisions.

InnaAtanasova reviewed (changes requested):
I'd suggest you revisit the focus concept:
1. In the specs I don't see focus on `down` state, but I might be wrong. In case I am wrong, then the color used in this case for sure is wrong (should be contrast).
<img width="216" alt="Screen Shot 2022-04-25 at 3 50 45 PM" src="https://user-images.githubusercontent.com/39598672/165166487-4b56413a-4789-4d19-8079-453b52e34543.png">
<img width="295" alt="Screen Shot 2022-04-25 at 3 50 14 PM" src="https://user-images.githubusercontent.com/39598672/165166490-274f51ac-f9af-42ef-a08d-f0f9f618cb75.png">
 
2. This is the case with Horizon (styles vs specs)
<img width="170" alt="Screen Shot 2022-04-25 at 3 52 16 PM" src="https://user-images.githubusercontent.com/39598672/165166730-842aa496-0d7e-455c-9cee-08af22427c2d.png">
<img width="307" alt="Screen Shot 2022-04-25 at 3 51 56 PM" src="https://user-images.githubusercontent.com/39598672/165166734-c53311ff-6998-46dd-a749-0046cd8bf4c9.png">

3. The offset should be double-checked. In h
<img width="317" alt="Screen Shot 2022-04-25 at 3 35 40 PM" src="https://user-images.githubusercontent.com/39598672/165166893-ec353083-4531-4e07-9326-9a664f2feeb5.png">
<img width="667" alt="Screen Shot 2022-04-25 at 3 33 39 PM" src="https://user-images.githubusercontent.com/39598672/165166897-647bbbd9-7190-4f9e-be6e-d1223a45ebf2.png">
<img width="562" alt="Screen Shot 2022-04-25 at 3 33 33 PM" src="https://user-images.githubusercontent.com/39598672/165166898-ac1e54a5-8735-4efe-bcab-58264571f9fe.png">
igh contrast themes it's really off:

platon-rov commented:
> In the specs I don't see focus on down state, but I might be wrong. In case I am wrong, then the color used in this case for sure is wrong (should be contrast).

It's the same as for the toggle button, fixed.

> This is the case with Horizon

When we're pressing down the mouse button on the button element it gets both states, focus & active, it's intended behavior. That's why we see that focus ring. To change it a big amount of work should be done, also for many components, cause we have the same behavior everywhere. I'm not sure we should fix it in scope of this pr.

> The offset should be double-checked. In high contrast themes it's really off

We have `outline-offset: -0.1875rem;` for both - fiori & horizon, although we changed this value in horizon to `0.25rem` to get our visual closer to the screenshots. Probably we should raise that topic with the ui team, why when we're using variables correctly we end up with the visuals that don't match the screenshots in the wiki. 

About this exact example & split button

<img width="146" alt="image" src="https://user-images.githubusercontent.com/20265336/165350229-3738d5df-0241-4fe6-aed1-1119db4803fa.png">

It won't be ideal unfortunately. You can see that we have single border between the buttons, thou there should be double border (border-right of the left button and border-left of the right button) and we cannot set `outline-offset-left` for the right button, we just can manage the whole `outline-offset` and if we change it - it would be broken on the other three sides. Probably, as an option we can draw outline via the pseudo elements (before/after) in this exact case, however it's not the correct solution and a lot of hardcode will come. What do you think?

g-cheishvili commented on src/styles/button.scss:
```diff
@@ -60,215 +46,32 @@ $block: #{$fd-namespace}-button;
 
-  &.#{$block}--transparent {
-    @include transparentHover();
+  * + * {
```
```suggestion
  *:not(:first-child) {
```
It's a longer, but easier to grasp

g-cheishvili reviewed (approved):
Once @InnaAtanasova 's notes are applied I do not have anything against this PR. Code is VERY good!

droshev commented:
> https://deploy-preview-3328--fundamental-styles.netlify.app

We are not changing the design and behaviour but the implementation, so one of the expectations is we won't have regression with the current visuals/chromatic(excluding when we fix bugs)
